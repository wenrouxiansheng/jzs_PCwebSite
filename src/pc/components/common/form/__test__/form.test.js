import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Form from "../index";

describe("PC:<Form />", () => {
  it("render", () => {
    const wrapper = renderer.create(<Form />);
    let tree = wrapper.toJSON();

    //渲染测试
    expect(tree).toMatchSnapshot();
  });

  it("submit event", () => {
    const wrapper = mount(<Form />);
    wrapper.find("input#username").simulate("change", {
      target: {
        value: "13",
      },
    });
    wrapper.find("input#phone").simulate("change", {
      target: {
        value: "13161686315",
      },
    });
    wrapper.find("input#course").simulate("click");
    wrapper.find(".ant-select-item").at(1).simulate("click");
    console.log(wrapper.find("input#course").text())
    // wrapper.find("input#username").simulate("change", {
    //   target: {
    //     value: "13",
    //   },
    // });
  });
});
