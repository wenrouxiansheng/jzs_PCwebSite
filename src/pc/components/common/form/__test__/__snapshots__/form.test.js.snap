// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PC:<Form /> render 1`] = `
<div
  className="formContainer"
  style={
    Object {
      "background": "url(test-file-stub) no-repeat center center",
    }
  }
>
  <div
    className="container"
  >
    <form
      className="ant-form ant-form-horizontal"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <h1>
        报名参加桔子树
      </h1>
      <h3>
        免费试听课程
      </h3>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="username"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="姓名（必填）"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="phone"
                maxLength="11"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="手机号码（必填）"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                className="ant-select ant-select-single ant-select-show-arrow"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
              >
                <div
                  className="ant-select-selector"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                >
                  <span
                    className="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="course_list_0"
                      aria-autocomplete="list"
                      aria-controls="course_list"
                      aria-haspopup="listbox"
                      aria-owns="course_list"
                      autoComplete="off"
                      className="ant-select-selection-search-input"
                      id="course"
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onPaste={[Function]}
                      readOnly={true}
                      role="combobox"
                      style={
                        Object {
                          "opacity": 0,
                        }
                      }
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    className="ant-select-selection-placeholder"
                  >
                    请选择课程
                  </span>
                </div>
                <span
                  aria-hidden={true}
                  className="ant-select-arrow"
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    className="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                className="ant-select ant-select-single ant-select-show-arrow"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
              >
                <div
                  className="ant-select-selector"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                >
                  <span
                    className="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="school_list_0"
                      aria-autocomplete="list"
                      aria-controls="school_list"
                      aria-haspopup="listbox"
                      aria-owns="school_list"
                      autoComplete="off"
                      className="ant-select-selection-search-input"
                      id="school"
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onPaste={[Function]}
                      readOnly={true}
                      role="combobox"
                      style={
                        Object {
                          "opacity": 0,
                        }
                      }
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    className="ant-select-selection-placeholder"
                  >
                    请选择校区
                  </span>
                </div>
                <span
                  aria-hidden={true}
                  className="ant-select-arrow"
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    className="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        type="submit"
      >
        立即申请
      </button>
    </form>
  </div>
</div>
`;
